File: C++/Easy/108. 将有序数组转换为二叉搜索树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        int len=nums.size();
        if(len==0)
            return nullptr;
        if(len==1)
        {
            TreeNode* root= new TreeNode(nums[0]);
            return root;
        }
        int root_ind=(len%2==0)?len/2-1:len/2;
        TreeNode* root=new TreeNode(nums[root_ind]);
        vector<int> leftnums(nums.begin(),nums.begin()+root_ind);
        vector<int> rightnums(nums.begin()+root_ind+1,nums.end());
        root->left=sortedArrayToBST(leftnums);
        root->right=sortedArrayToBST(rightnums);
        return root;
    }
};

File: python/Easy/108. 将有序数组转换为二叉搜索树.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
        def make_tree(left_ind, right_ind) -> TreeNode:
            if left_ind > right_ind:
                return None
            mid_ind = (left_ind + right_ind) // 2
            root = TreeNode(nums[mid_ind])
            root.left = make_tree(left_ind, mid_ind-1)
            root.right = make_tree(mid_ind+1, right_ind)
            return root
        return make_tree(0, len(nums)-1)